### DISCUSSION TITLE
Getting exception as Invalid arguments  were passed to DockerSwarmOperator 

### DISCUSSION THREAD

**Discussion by Saga305** [apache/airflow#20885] [Q&A] [answered]
I am getting exception from baseoperator. Py as Invalid arguments were passed to DockerSwarmOperator while using Dockerswarmoperator. Any idea to fix this? 

I have added placement constraints and for that I'm getting exception. 

Syntax used for constraints as below. 

From docker.types import Placemnt
placement=[Placement(constraints=["node.labels.test==true"])]

Or 

placement=Placement(constraints=["node.labels.test==true"])

Exception snippets:

Broken DAG: [/opt/airflow/dags/bfp-ah-process.py] Traceback (most recent call last):

  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 188, in apply_defaults

    result = func(self, *args, **kwargs)

  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 537, in __init__

    "arguments were:\n**kwargs: {k}".format(c=self.__class__.__name__, k=kwargs, t=task_id),

airflow.exceptions.AirflowException: Invalid arguments were passed to DockerSwarmOperator (task_id: test). Invalid arguments were:

**kwargs: {'placement': [{'Constraints': ['node.labels.test==true']}]}

**Comment by jedcunningham** [comment, marked as answer]
Are you on version `2.3.0` or `2.4.0` of `apache-airflow-providers-docker`? Support for it was added in `2.3.0`:

https://airflow.apache.org/docs/apache-airflow-providers-docker/stable/index.html#changelog
